### DISCUSSION TITLE
/etc/localtime

### DISCUSSION THREAD

**Discussion by zadigus** [wolfi-dev/os#24820] [General]
Hi,

It appears one of the latest wolfi builds has introduced an empty file `/etc/localtime` which breaks our python applications with the following error:
 ```
File "/venv/lib/python3.11/site-packages/tzlocal/unix.py", line 219, in get_localzone
    _cache_tz = _get_localzone()
                ^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/tzlocal/unix.py", line 190, in _get_localzone
    tz = zoneinfo.ZoneInfo.from_file(tzfile, key="local")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/zoneinfo/_common.py", line 28, in load_data
    header = _TZifHeader.from_file(fobj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/zoneinfo/_common.py", line 146, in from_file
    raise ValueError("Invalid TZif file: magic not found")
ValueError: Invalid TZif file: magic not found
```

The problem comes from the fact that the file exists and is empty, hence carrying the wrong data. Is that known to the wolfi maintainers team? Was that intended? Will that be solved?

**Comment by rawlingsj** [comment]
Hi thanks for raising this and sorry about that.  We reverted a change last night which seems related https://github.com/wolfi-dev/os/pull/24808. Would you mind taking a look to see if this has fixed the issue please?

**Reply by zadigus** [reply]
it looks good; we'll do some more detailed testing throughout the day but as far as I can tell from the first tests it looks good

**Reply by zadigus** [reply]
I confirm it's all good.

**Reply by rawlingsj** [reply]
great to hear :)
